File: python_projeto_02_ampliar_reduzir_interpolacao_bilinear/projeto_02_reduzir.py
'''
    Escreva um programa capaz de ampliar e reduzir uma imagem por replicação
    de pixel. Assuma que os fatores de ampliação/redução sejam inteiros;
'''

from PIL import Image

# Path da imagem
src_image = "../img/Fig0219.tif"

# Imagem de entrada
in_img = Image.open(src_image)

# Tamanho da imagem LINHAS, COLUNAS
M, N = in_img.size

# Carrega na variavel a imagem de entrada
in_pixel = in_img.load()

# Entrada da redução
reducao = int(input("Digite quanto vc quer que diminua"))

MM = round(M / reducao)
print(str(MM))
NN = round(N / reducao)
print(str(NN))

k = 1
l = 1

#TodO..........
# Imagem de saida
out_img = Image.new("L", (MM,NN))
out_pixel = out_img.load()

for m in range(MM):
    for n in range(NN):
        out_pixel[m,n] = in_pixel[k,l]
        l = l + reducao
    l = 1
    k = k + reducao

out_img.save('out_image_XXX_python.png', 'png')

# Mostra imagem de saida
out_img.show()

File: matlab_projeto_02_ampliar_reduzir_ replicação_pixel/projeto_02_reduzir.m
% Escreva um programa capaz de ampliar e reduzir uma imagem por replicação
% de pixel. Assuma que os fatores de ampliação/redução sejam inteiros;

% Caminho do arquivo que deverá ser modificado
path = './img/Fig0219.tif';

I = imread(path);

[M, N] = size(I);

% Inteiro de entrada
f=input('Entre com o fator de redução da imagem: ');

M1 = round(M / f);
fprintf('%d',M1);
N1 = round(N / f);
fprintf('%d',N1);

k=1;
l=1;

% imshow(I, []);
% title('Imagem original');

% Criando uma imagem de saida do mesmo tamanho da de entrada
I_out = zeros(M1, N1);

for m = 1:M1
    for n = 1:N1
        I_out(m,n) = I(k,l);
        l=l+f;
    end
    l=1;
    k=k+f;
end

imwrite(I_out, 'matlab_imagem_64', 'jpg');

figure();
imshow(I_out, []);
